server/utils.go

```diff
@@ -49,18 +51,48 @@ func PostJSONWithTimeout(ctx context.Context, url string, v interface{}) error {
 
 	resp, err := client.Do(req)
 	if err != nil {
-		return fmt.Errorf("failed to POST to %s: %s, request-size=%d", url, err, len(jsonBytes))
+		return fmt.Errorf("failed to POST to %s: %s, request-size=%d", maskSecretURLParams(url), err, len(jsonBytes))
 	}
 	defer resp.Body.Close()
 
 	if !httpSuccessStatus(resp.StatusCode) {
 		body, _ := ioutil.ReadAll(resp.Body)
-		return fmt.Errorf("error posting to %s: %d. %s", url, resp.StatusCode, string(body))
+		return fmt.Errorf("error posting to %s: %d. %s", maskSecretURLParams(url), resp.StatusCode, string(body))
 	}
 
 	return nil
 }
 
+// maskSecretURLParams masks URL query values if the query param name includes "secret", "token",
+// "key", "password". It accepts a raw string and returns a redacted string if the raw string is
+// URL-parseable. If it is not URL-parseable, the raw string is returned unchanged.
+func maskSecretURLParams(rawURL string) string {
+	matchKeyword := func(k string) bool {
+		kws := []string{"secret", "token", "key", "password"}
+		for _, kw := range kws {
+			if strings.Contains(strings.ToLower(k), kw) {
+				return true
+			}
+		}
+		return false
+	}
+
+	u, err := url.Parse(rawURL)
+	if err != nil {
+		return rawURL
+	}
+
+	q := u.Query()
+	for k := range q {
+		if matchKeyword(k) {
+			q[k] = []string{"MASKED"}
+		}
```

Comment:
can we use a map to reduce the code and algorithmic complexity of this?

```go
	kws := map[string]struct{}{
		"secret":   struct{},
		"token":    struct{},
		"key":      struct{},
		"password": struct{},
	}

	u, err := url.Parse(rawURL)
	if err != nil {
		return rawURL
	}

	q := u.Query()

	for k := range q {
		if _, ok := kws[k]; ok {
			q[k] = []string{"MASKED"}
		}
	}
```

Reply:
I thought about that but I thought it was worth a little added complexity to get substring matching (e.g., "webhookSecret" or "webhook_secret").

Reply:
Sorry, I missed that! I would advocate to not use `strings.Contains` at all, **but** worst that can happen is that we mask something that we don't want to mask, so I'm not feeling strongly about that.

As a tiny nit, feel free to ignore: can we do `strings.ToLower` only once instead of on each loop?

Reply:
> As a tiny nit, feel free to ignore: can we do strings.ToLower only once instead of on each loop?

Yes! I caught that too and was going to push a change :)

Reply:
> I would advocate to not use strings.Contains at all, but worst that can happen is that we mask something that we don't want to mask

I debated the risk of over-masking and also lower performance of the `strings.Contains` algorithm. I figured we could revisit in the future if either of those became a problem and look into alternative approaches then.